### ISSUE TITLE
Change master branch to main branch

### ISSUE THREAD

**Issue by vinferrer** [physiopy/phys2bids#356]
I'm opening this discussion because I think that there are several reasons to do this change:
1. Avoid racists and discriminative language implied in the ``master/slave`` words.
2. Since ``main`` is becoming a default in new repos, we should adopt it.

## Outstanding questions

    - I found a tutorial on how to do it, thought i don't have privileges. Who wanna take care of it: 
    - https://stevenmortimer.com/5-steps-to-change-github-default-branch-from-master-to-main/

**Comment by eurunuela** [comment]
Well, I'd say it's more about when we want to make the change.

As you say, it's a simple change to make but every phys2bids developer should be noticed first.

I think the best we can do is set a date and announce in a couple of weeks time to make sure we all know about the transition.

**Comment by vinferrer** [comment]
Sure, that's why i oppened an issue

**Comment by tsalo** [comment]
This came up in `tedana` as well (see https://github.com/ME-ICA/tedana/issues/619). Based on [this post](https://github.com/github/renaming/tree/4c8e50916922d98617e1c306afc8a8fcbf036a52#later-this-year-seamless-move-for-existing-repositories-), I think that GitHub will roll out changes before year's end to make the transfer easier. I would wait until GitHub announces those changes.

**Comment by vinferrer** [comment]
oh, fair enough

**Comment by RayStick** [comment]
Yes I am in full support of changing this but happy to wait a short while if that is sensible for logistics.

**Comment by smoia** [comment]
Hi all,
I fully support the change only if we use the name _maegister_ or _magister_ (how cool is to bring back some Latin here and there).
Jokes aside, I'm sorry to be the voice out of the choir, but my support of changing names comes with some conditions.
I definitely won't be against this process (after all, it's not a bad process), but I would like to invite you to think about what we're doing and why we're doing it.

On one side, we have contributors that might need some help in getting this change done (if they want to) or in any case might have to adjust their configuration to keep working with `upstream`. On the other, we have some configurations of our own to adapt to the change.
In addition to that, I have the bad feeling that this is more of an image stunt rather than a real, felt, change for the better. So, since I'd like to put action where our mouth is, I would propose to actually find a way to actively support reduction of any form of slavery (or simil-slavery) and partake somehow as `physiopy`.

For that, I would propose that we nominate a person or a team responsible for taking care of this issue. On one side, this person(s) might help and ease the transition (if we see that GitHub didn't ease the process enough), similarly to what @jbteves is volunteering to do in ME-ICA/tedana#619. This means taking care of helping people in changing configurations and adapting our workflows to work with `main`. On the other side, this person(s) could look into and find a good and doable action that we can implement as a community. It could be a statement, a tweet/github challenge, a donation for a charity, or a possibly better idea.

We can discuss this issue further next week during our meeting (agenda coming in the upcoming two days).
What do you all think?

**Comment by RayStick** [comment]
Thanks for your thoughts Stefano. I will have a think about them, and bring thoughts to the next meeting, which might be an easier/better forum to discuss.

**Comment by tsalo** [comment]
I can't speak to the larger proposed changes, but it looks like the steps for renaming the `master` branch have recently been improved (see [this post](https://github.com/github/renaming/tree/040636ea50e945875c70cc964d586f645e9bd810#renaming-existing-branches)), so it should be much easier to do now.

**Comment by RayStick** [comment]
As per[ this post ](https://github.com/github/renaming/tree/040636ea50e945875c70cc964d586f645e9bd810#renaming-existing-branches)(updated since Taylor shared it I think) it seems you can now simply rename the branch from master to main. I did that for one of my own repos on GitHub, and this popped up:

<img width="454" alt="Screen Shot 2021-05-06 at 5 13 03 PM" src="https://user-images.githubusercontent.com/50215726/117372140-add5ef80-ae8e-11eb-93d9-6caa1f069b42.png">

I followed those instructions and all seemed to work fine. Though this was on a repo I don't really use. 
It says that these things will happen if we rename from master to main:

- Re-target any open pull requests
- Update any draft releases based on the branch
- Move any branch protection rules that explicitly reference the old name
- Update the branch used to build GitHub Pages, if applicable
- Show a notice to repository contributors, maintainers, and admins on the repository homepage with instructions to update local copies of the repository
- Show a notice to contributors who git push to the old branch
- Redirect web requests for the old branch name to the new branch name
- Return a "Moved Permanently" response in API requests for the old branch name

Do any of those things look like they'd pose a problem?

**Comment by eurunuela** [comment]
We will have to make sure that the CI and docs point to `main` rather than `master`. Other than that, I can't think of any other issue we could face when changing.
